cqrs/core/event-store-adapters/MemoryAdapter.js

```diff
@@ -0,0 +1,63 @@
+import {AbstractEventStoreAdapter} from '../abstraction/AbstractEventStoreAdapter';
+
+class MemoryAdapter extends AbstractEventStoreAdapter {
+
+    constructor (eventBus) {
+        super(eventBus);
+        this._aggregates = {};
+        this._events = [];
+        this._eventBus = eventBus;
+    }
+
+    readAllEvents (uuid) {
+        if (!this.isAggregateRegistered(uuid)) return null;
+        return this._getAggregateEvents(uuid);
+    }
+
+    writeEvents (uuid, originalVersion = 0, events) {
+        return new Promise ((resolve, reject) => {
+            if (!this.isAggregateRegistered(uuid)) this._registerNewAggregate(uuid);
+
+            this._getAggregate(uuid).then((aggregate) => {
+                let version = aggregate.version;
+                if (version !== originalVersion) throw new Error("Aggregate's version mismatch");
+
+                events.forEach(event => {
+                    version++;
+                    this._updateAggregate(uuid, {version});
+                    this._eventBus.push(event);
+                });
+                this._events.push.apply(events.map(event => {...event, ag_uuid: uuid}));
```

Comment:
Actually, I wasn't compiling this.
But this construction will work with https://babeljs.io/docs/plugins/transform-object-rest-spread/ plugin
[Object Rest/Spread Proposal](https://github.com/sebmarkbage/ecmascript-rest-spread)